proxy/src/main/java/com/velocitypowered/proxy/connection/client/ClientPlaySessionHandler.java

```diff
@@ -229,9 +230,11 @@ public boolean handle(PluginMessage packet) {
         player.getKnownChannels().removeAll(PluginMessageUtil.getChannels(packet));
         backendConn.write(packet.retain());
       } else if (PluginMessageUtil.isMcBrand(packet)) {
+        String brand = PluginMessageUtil.readBrandMessage(packet.content());
+        server.getEventManager().fire(new PlayerClientBrandEvent(player, brand));
         player.setClientBrand(PluginMessageUtil.readBrandMessage(packet.content()));
```

Comment:
```suggestion
        player.setClientBrand(brand);
```
Strings are immutable.